src/main/java/iudx/resource/server/database/elastic/ConvertElasticResponseToJson.java

```diff
@@ -0,0 +1,67 @@
+package iudx.resource.server.database.elastic;
+
+import co.elastic.clients.elasticsearch.core.search.Hit;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class ConvertElasticResponseToJson extends AbstractConvertElasticSearchResponse {
+  private final FileWriter fileWriter;
+  //    private static final Logger LOGGER =
+  // LogManager.getLogger(ConvertElasticResponseToJSON.class);
+
+  /**
+   * Writes ElasticSearch response batch response into a JSON File
+   *
+   * @param file File to write JSON response
+   */
+  public ConvertElasticResponseToJson(File file) {
+    super(file);
+    try {
+      this.fileWriter = new FileWriter(file);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void write(List<Hit<ObjectNode>> searchHits) {
+    try {
+      fileWriter.write('[');
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void append(List<Hit<ObjectNode>> searchHits, boolean isLastRecord) {
+    try {
+      fileWriter.write(']');
```

Comment:
It should check for `if(isLastRecord)` and only close the file write. else call `append(searchHits,false)`

Reply:
Will do

Reply:
Changed it here : [Refactor files](https://github.com/datakaveri/iudx-resource-server/pull/425/commits/3f2f16ce5bedb80fe7235b2d794c3393641dcc2c)